feat(nhcb): add support for NHCB over RW1/RW2 (#11143)
grafana/mimir: docs/sources/mimir/manage/mimir-runbooks/_index.md

where the number of bucket counts does not agree with the number of buckets encoded in the bucket spans.

{{< admonition type="note" >}}
Series containing these samples are skipped during ingestion. Valid series in the same request are ingested.
{{< /admonition >}}

{{< admonition type="note" >}}
When you configure `-ingester.error-sample-rate` to a value of `N` that is greater than `0`, only every `Nth` invalid native histogram error is logged.
{{< /admonition >}}

### err-mimir-native-histogram-custom-buckets-mismatch

This non-critical error occurs when Mimir receives a write request that contains a sample that is a native histogram
with custom buckets where more buckets are present than the upper bound is specified for. In technical terms, the

> number of bucket values from spans is higher than the number of custom values.

Reviewer: Should we use code format for `spans`?

{{< admonition type="note" >}}
Series containing these samples are skipped during ingestion. Valid series in the same request are ingested.
{{< /admonition >}}

{{< admonition type="note" >}}
When you configure `-ingester.error-sample-rate` to a value of `N` that is greater than `0`, only every `Nth` invalid native histogram error is logged.
{{< /admonition >}}

### err-mimir-native-histogram-custom-buckets-invalid

This non-critical error occurs when Mimir receives a write request that contains a sample that is a native histogram
with custom buckets where the bucket boundaries are not in ascending order.

{{< admonition type="note" >}}
Series containing these samples are skipped during ingestion. Valid series in the same request are ingested.
{{< /admonition >}}

{{< admonition type="note" >}}
When you configure `-ingester.error-sample-rate` to a value of `N` that is greater than `0`, only every `Nth` invalid native histogram error is logged.
{{< /admonition >}}

### err-mimir-native-histogram-custom-buckets-infinite

This non-critical error occurs when Mimir receives a write request that contains a sample that is a native histogram

> with custom buckets where one of the boundaries is +Inf, which should not be specified as it is the default implicit

Reviewer: ```suggestion
with custom buckets where one of the boundaries is `+Inf`, which should not be specified as it is the default implicit
```

When `-ingester.error-sample-rate` is configured to a value greater than `0`, invalid native histogram errors are logged only once every `-ingester.error-sample-rate` times.
{{< /admonition >}}

### err-mimir-native-histogram-custom-buckets-mismatch

This non-critical error occurs when Mimir receives a write request that contains a sample that is a native histogram

> with custom buckets where more buckets are present than the upper bound is specified for. In technical terms: the
> number of bucket values from spans is higher than the number of custom values.

Reviewer: ```suggestion
with custom buckets where more buckets are present than the upper bound is specified for. In technical terms, the
number of bucket values from spans is higher than the number of custom values.
```

{{< admonition type="note" >}}

> The series containing such samples are skipped during ingestion, and valid series within the same request are ingested.

Reviewer: ```suggestion
Series containing these samples are skipped during ingestion. Valid series in the same request are ingested.
```

{{< /admonition >}}

{{< admonition type="note" >}}

> When you configure `-ingester.error-sample-rate` to a value greater than `0`, invalid native histogram errors are logged only once every `-ingester.error-sample-rate` times.

Author: @tacole02 how about:
```suggestion
When you configure `-ingester.error-sample-rate` to a value of `N` that is greater than `0`, only every `Nth` invalid native histogram error is logged.
```

Author: Note: these two notes are copy pasted from existing docs they occur dozens of times.

Reviewer: I think that sounds clear and more specific than the original! Thanks, @krajorama !

> {{< admonition type="note" >}}
> When `-ingester.error-sample-rate` is configured to a value greater than `0`, invalid native histogram errors are logged only once every `-ingester.error-sample-rate` times.

Reviewer: ```suggestion
{{< admonition type="note" >}}
When you configure `-ingester.error-sample-rate` to a value greater than `0`, invalid native histogram errors are logged only once every `-ingester.error-sample-rate` times.
```

Reviewer: > invalid native histogram errors are logged only once every `-ingester.error-sample-rate` times.

This is confusing to me. Could we clarify how often the errors are logged? Is it that they're logged at the rate set in `-ingester.error-sample-rate`?

{{< /admonition >}}

### err-mimir-native-histogram-custom-buckets-invalid

This non-critical error occurs when Mimir receives a write request that contains a sample that is a native histogram
with custom buckets where the bucket boundaries are not in ascending order.

{{< admonition type="note" >}}

> The series containing such samples are skipped during ingestion, and valid series within the same request are ingested.

Reviewer: ```suggestion
Series containing these samples are skipped during ingestion. Valid series in the same request are ingested.
```

{{< /admonition >}}

{{< admonition type="note" >}}

> When `-ingester.error-sample-rate` is configured to a value greater than `0`, invalid native histogram errors are logged only once every `-ingester.error-sample-rate` times.

Reviewer: ```suggestion
When you configure `-ingester.error-sample-rate` to a value greater than `0`, invalid native histogram errors are logged only once every `-ingester.error-sample-rate` times.
```

Reviewer: Same comment as above re. the rate of error logging.

{{< /admonition >}}

### err-mimir-native-histogram-custom-buckets-infinite

This non-critical error occurs when Mimir receives a write request that contains a sample that is a native histogram
with custom buckets where one of the boundaries is +Inf, which should not be specified as it is the default implicit
last boundary.

{{< admonition type="note" >}}

> The series containing such samples are skipped during ingestion, and valid series within the same request are ingested.

Reviewer: ```suggestion
Series containing these samples are skipped during ingestion. Valid series in the same request are ingested.
```

{{< /admonition >}}

{{< admonition type="note" >}}

> When `-ingester.error-sample-rate` is configured to a value greater than `0`, invalid native histogram errors are logged only once every `-ingester.error-sample-rate` times.

Reviewer: ```suggestion
When you configure `-ingester.error-sample-rate` to a value greater than `0`, invalid native histogram errors are logged only once every `-ingester.error-sample-rate` times.
```

Reviewer: Same comment as above.

On a per-tenant basis, you can fine tune the tolerance by configuring the `creation_grace_period` option.

{{< admonition type="note" >}}

> Invalid exemplars are skipped during ingestion. Valid exemplars in the same request are ingested.

Author: note to reviewers: this is not just style change, but also a bugfix. We check exemplars after samples are ingested, so the statement was wrong.
